### PR TITLE
Generate argument parser from config setting list

### PR DESCRIPTION
**Pull request by keszybz**

### CODE REVIEW

**Comment by keszybz** [comment]
CI / integration-test (ubuntu, directory) failed on some apt network error. I'll restart it.

File: mkosi/config.py [hunk 1]
```diff
@@ -742,465 +751,681 @@ class MkosiConfigParser:
...
         ),
         MkosiConfigSetting(
             dest="root_password",
+            metavar="PASSWORD",
             section="Content",
             parse=config_parse_root_password,
             paths=("mkosi.rootpw",),
             path_read_text=True,
             path_secret=True,
+            help="Set the root's password",
```
Comment (behrmann) [inline mkosi/config.py:1213]: the combination of article and possessive marker looks weird, I think either of these alternatives would work
```suggestion
            help="Set the root password",
```
or
```suggestion
            help="Set root's password",
```
or
```suggestion
            help="Set the root account's password",
```
or
```suggestion
            help="Set the root user's password",
```

My preference would be the first one.

File: mkosi/config.py [hunk 2]
```diff
@@ -742,465 +751,681 @@ class MkosiConfigParser:
...
         ),
         MkosiConfigSetting(
             dest="root_shell",
+            metavar="SHELL",
             section="Content",
             parse=config_parse_string,
+            help="Set the root's shell",
```
Comment (behrmann) [inline mkosi/config.py:1220]: Same, and also in the commit message.

File: mkosi/config.py [hunk 3]
```diff
@@ -1422,545 +1647,27 @@ def create_argument_parser(self) -> argparse.ArgumentParser:
...
-        )
-        group.add_argument(
-            "--sign",
-            help="Write and sign SHA256SUMS file",
-            metavar="BOOL",
-            nargs="?",
-            action=action,
-        )
-        group.add_argument("--key", help="GPG key to use for signing", action=action)
+            group.add_argument(    # type: ignore
```
Comment (behrmann) [inline mkosi/config.py:1660]: If you still have the warning in some log adding that to the commit message would be nice.

**Review by behrmann — COMMENTED** [review]
Excellent change, some minor nits.

**Comment by behrmann** [comment]
The CI failure is unrelated.

**Review by DaanDeMeyer — APPROVED** [review]

File: mkosi/config.py (hunk 1 above)
Comment (keszybz) [reply mkosi/config.py:1213]: Hmm, the first one is how we'd usually write it. But "set zbyszek password" looks very strange, and really "root" is just the account name here. What about "set password for root"? I think that avoids the issue nicely.

**Comment by keszybz** [comment]
Updated with "change the password|shell for root" (split out into a separate commit since with a whole paragraph of musings for the gramatically inclined), and the amended message in the last commit.

**Pull request merged**